python/dgl/distributed/kvstore.py

```diff
@@ -0,0 +1,1107 @@
+        # Used for barrier() API on KVClient
+        self._barrier_count = 0
+        # Basic information
+        self._server_id = server_id
+        self._server_namebook = rpc.read_ip_config(ip_config)
+        self._machine_id = self._server_namebook[server_id][0]
+        self._group_count = self._server_namebook[server_id][3]
+        self._part_id = self._machine_id
+        self._num_clients = num_clients
+        # TODO(chao) : remove tmp file using new shared-tensor API
+        self._open_file_list = []
+        # push and pull handler
+        self._push_handler = self._default_push_handler
+        self._pull_handler = self._default_pull_handler
+
+    def __del__(self):
+        """Finalize KVServer
+        """
+        # Delete temp file when kvstore service is closed
+        # TODO(chao) : remove tmp file using new shared-tensor API
+        for file in self._open_file_list:
+            if (os.path.exists(file)):
+                os.remove(file)
+
+     @property
+     def server_id(self):
+         return self._server_id
+
+     @property
+     def barrier_count(self):
+     	return self._barrier_count
+
+     def incre_barrier_count(self):
+        self._barrier_count +=1
+
+     @property
+     def num_clients(self):
+     	return self._num_clients
+     
+     @property
+     def data_store(self):
+         return self._data_store
+
+     @property
+     def part_policy(self):
+         return self._part_policy
+     
+     @property
+     def push_handler(self):
+         return self._push_handler
+
+     @property
+     def pull_handler(self):
+         return self._pull_handler
+
+     @pull_handler.setter
+     def pull_handler(self, pull_handler):
+         self._pull_handler = pull_handler
+
+     @push_handler.setter
+     def push_handler(self, push_handler):
+         self._push_handler = push_handler
+
+     def is_main_server(self):
+         """Return True is current server is not a backup-server.
+         """
+         if self._server_id % self._group_count == 0:
+             return True
+         return False
+
+    def init_data(self, name, data_tensor=None):
+        """Init data tensor on kvserver.
+
+        Parameters
+        ----------
+        name : str
+            data name
+        data_tensor : tensor
+            If the data_tensor is None, KVServer will read shared-memory by name.
+        """
+        # TODO(chao) : remove tmp file using new shared-tensor API
+        assert len(name) > 0, 'name cannot be empty.'
+        if data_tensor is not None: # Create shared-tensor
+            data_type = get_type_str(F.dtype(data_tensor))
+            shared_data = empty_shared_mem(name+'-kvdata-', True, data_tensor.shape, data_type)
+            dlpack = shared_data.to_dlpack()
+            self._data_store[name] = F.zerocopy_from_dlpack(dlpack)
+            self._data_store[name][:] = data_tensor[:]
+            self._write_data_meta_to_file(name+'-kvmeta-'+str(self._machine_id), data_tensor)
```

Comment:
you still write metadata to files?

Reply:
It will be replaced by new shared-memory api